frontend/services/entities/scripts.ts

```diff
@@ -179,10 +193,37 @@ export interface IScriptBatchSummariesQueryKey
   scope: "script_batch_summaries";
 }
 
-export interface IScriptBatchSummariesResponse {
+export interface IScriptBatchSummariesResponse
+  extends ListEntitiesResponseCommon {
   batch_executions: IScriptBatchSummaryV2[] | null; // should not return `null`, but API currently does sometimes. Remove this option when it's fixed.
-  meta: PaginationMeta;
-  count: number;
+}
+
+export type ScriptBatchHostsOrderKey = "display_name" | "script_executed_at";
+export interface IScriptBatchHostResultsParams extends PaginationParams {
+  batch_execution_id: string;
+  status: ScriptBatchHostStatus;
+  order_key: ScriptBatchHostsOrderKey;
+  order_direction: OrderDirection;
+}
+
+export interface IScriptBatchHostResultsQueryKey
+  extends IScriptBatchHostResultsParams {
+  scope: "script_batch_host_results";
+}
+
+export interface IScriptBatchHostResult {
+  id: number;
+  display_name: string;
+  script_status: ScriptBatchHostStatus;
+  script_execution_id: string | null; // if status === pending, this may be `null` or contain a value dependending on whether the script exectution is this hosts next scheduled activity on the server
```

Comment:
Maybe the API should not return a script execution ID for pending executions? But you're already handling it, and there could be some value in it in the future.